oauthproxy.go

```diff
@@ -971,10 +971,7 @@ func (p *OAuthProxy) Proxy(rw http.ResponseWriter, req *http.Request) {
 
 		logger.Printf("No valid authentication in request. Initiating login.")
 		if p.SkipProviderButton {
-			// start OAuth flow, but only with the default login URL params - do not
-			// consider this request's query params as potential overrides, since
-			// the user did not explicitly start the login flow
-			p.doOAuthStart(rw, req, nil)
+			p.doOAuthStart(rw, req, req.URL.Query())
```

Comment:
How has this been tested? Could you add a test case for this change in behaviour?

Reply:
I've only ran this locally in my own container unfortunately. 

I'm not too experienced in go as well, so I'm currently a bit clueless atm for writing unit tests. Is there any example that I couple reference / copy?

Reply:
@tuunit I've updated the changelog

Reply:
Hi @flashpaper12, 

have a look at the test cases in `oauthproxy_test.go`. Especially `TestApiRoutes(t *testing.T)`, `TestSignInPageSkipProvider`, `TestSignInPageInvalidQueryStringReturnsBadRequest` and the page initialization method `NewSignInPageTest.